components/payments.tsx

```diff
@@ -1,58 +1,137 @@
 import { format } from 'date-fns';
 import startCase from 'lodash/startCase';
 import { BsCheckCircleFill, BsReceipt } from 'react-icons/bs';
-
+import { ProductImage } from './product';
 export interface OrderStatusProps {
-  status: 'processing' | 'shipped' | 'delivered';
+  status: 'unknown' | 'in_transit' | 'delivered' | 'error';
   trackingNumber?: string | null;
 }
 
-export const OrderStatus: React.FC<OrderStatusProps> = ({ status, trackingNumber }) => {
-  const colorMap = {
-    processing: 'primary',
-    shipped: 'gray',
-    delivered: 'purple'
+export const OrderStatus = ({ status, trackingNumber }: OrderStatusProps) => {
+  const statusMap = {
+    unknown: {
+      text: 'Processing',
+      color: 'gray'
+    },
+    in_transit: {
+      text: 'Shipped',
+      color: 'green'
+    },
+    delivered: {
+      text: 'Delivered',
+      color: 'purple'
+    },
+    error: {
+      text: 'Error',
+      color: 'red'
+    }
   };
+
+  const statusObj = statusMap[status];
+
   return (
     <Flex sx={{ gap: '.5rem' }}>
-      <Badge py={1} px={2} backgroundColor={colorMap[status] ?? 'gray'}>
-        {startCase(status)}
+      <Badge py={1} px={2} backgroundColor={statusObj.color}>
+        {startCase(statusObj.text)}
       </Badge>
       {trackingNumber && (
-        <Badge py={1} px={2} backgroundColor="gray">
+        <Badge py={1} px={2} backgroundColor={statusObj.color}>
           {trackingNumber}
         </Badge>
       )}
     </Flex>
   );
 };
 
-const ordersFixture: OrderStatusProps[] = [
-  {
-    status: 'processing',
-    trackingNumber: null
-  },
-  {
-    status: 'shipped',
-    trackingNumber: 'Z0123456789R1337'
-  },
-  {
-    status: 'delivered'
-  }
-];
+export interface ProductLineItemProps {
+  id: string;
+  name: string;
+  description: string;
+  images?: string[];
+  amount: number;
+  currency: string;
+  quantity: number;
+}
+
+const ProductLineItem = ({ id, name, description, images, quantity, amount, currency }: ProductLineItemProps) => {
+  return (
+    <Card sx={{ height: '100%', cursor: 'pointer', p: 0 }}>
+      <NextLink href={`/product/${id}`} passHref>
+        <Grid gap={2} columns={['1fr 2fr 1fr']}>
+          <Box sx={{ textAlign: 'left' }}>
+            <Link sx={{ display: 'inline-block' }}>
+              <ProductImage maxHeight="60px" images={images} />
+            </Link>
+          </Box>
+          <Flex sx={{ flexDirection: 'column', justifyContent: 'center', alignItems: 'left' }}>
+            <Heading sx={{ margin: '0', fontSize: '1em', lineHeight: '1' }}>
+              <Link sx={{ color: '#333', ':hover': { color: 'primary' }, textDecoration: 'none' }}>
+                {quantity} x {name}
+              </Link>
+            </Heading>
+            <Paragraph sx={{ fontSize: '0.7rem', textAlign: 'initial' }}>
+              <Text>{description}</Text>
+            </Paragraph>
+          </Flex>
+          <Box sx={{ textAlign: 'right' }}>{formatPrice(currency, amount)}</Box>
```

Comment:
I'm seeing that this returns `NaN` for some line items

<img width="455" alt="Screen Shot 2022-04-08 at 1 10 23 PM" src="https://user-images.githubusercontent.com/6089482/162489936-11a6749b-b227-4c82-af8a-6a7c01c51802.png">

Reply:
You can reproduce by creating a new purchase that combines a subscription with a one-time purchase.

Reply:
This should be fixed, can you check again?

Reply:
Oops, had a bug there, try again in a minute.

Reply:
Looks good now!